sintr_analysis_server_example/lib/query/completion_metrics.dart

```diff
@@ -90,6 +105,37 @@ final completionReductionMerge = (Map results1, Map results2) {
 final OPEN_BRACE = '{'.codeUnitAt(0);
 final QUOTE = '"'.codeUnitAt(0);
 
+/// Update the calculated values in the SDK results map
+void updateCalculations(sdkResults) {
+  List<int> values = sdkResults[RESPONSE_TIMES];
+  sdkResults[AVE] = sdkResults[TOTAL] / values.length;
+  sdkResults[V90TH] = values[(values.length * (9 / 10)).floor()];
+  sdkResults[V99TH] = values[(values.length * (99 / 100)).floor()];
+  sdkResults[RESPONSE_TIME_CLUSTERS] = _clusterValues(values, [32]);
+  var counts = sdkResults[RESULT_COUNTS];
+  sdkResults[RESULT_COUNT_CLUSTERS] = _clusterValues(counts, [0, 1, 5, 50]);
+}
+
+/// Return a clustered map generated from the given [values]
```

Comment:
I think this is an 'allocation map' rather than a cluster map? Unless I'm misreading the algorithm you're using?